packages/cli/src/models/local/UnpackController.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import request from 'request-promise-native';
+import fs from 'fs-extra';
+
+import { Logger } from 'zos-lib';
+
+import stdout from '../../utils/stdout';
+import patch from '../../utils/patch';
+import cp from '../../utils/cp';
+
+const ora = patch('ora');
+const simpleGit = patch('simple-git/promise');
+
+interface KitConfig {
+  ignore: string[],
+  hooks: Object,
+}
+
+export default class UnpackController {
+  public async unpack(url: string, workingDirPath: string = '', config: KitConfig): Promise<void | never> {
+    if (!url) throw Error('A url must be provided.');
+    if (!config) throw Error('A config must be provided.');
+
+    const { exec } = cp;
+    const { readdir, remove, pathExists } = fs;
+
+    // because zos always spawns '.zos.lock' file
+    if ((await readdir(workingDirPath)).length > 1) throw Error('The directory must be empty');
+
+    try {
+      let spinner = ora(`Downloading kit from ${url}`).start();
+      const git = simpleGit(workingDirPath);
+      await git.init();
+      await git.addRemote('origin', url);
+      await git.pull('origin', 'stable');
+      spinner.succeed();
+
+      spinner = ora(`Unpacking kit`).start();
+      // always delete .git folder 
+      config.ignore.push(`.git`);
+      // delete all files/folders from ignore
+      const paths = config.ignore.map(file => path.join(workingDirPath, file));
+      const exists = await Promise.all(paths.map(path => pathExists(path)));
+      await Promise.all(exists.map((ext, i)=> ext ? remove(paths[i]) : null ).filter(Boolean));
+      
+      // run kit commands like `npm install`
+      await exec(config.hooks['post-unpack']);
+      spinner.succeed();
+
+      stdout(`The kit is ready to use. Amazing!`);
+      stdout(`Please, continue at ${url}`);
+    } catch(e) {
+      // TODO: remove all files from directory on fail except .zos.lock
+      e.message = `Failed to download and unpack kit from ${url}. Details: ${e.message}`;
+      throw e;
```

Comment:
What would happen with `ora` if something goes wrong inside the `try` statement? 
I would add a `spinner.fail()` inside the `catch` statement. Otherwise, the spinner animation will 

Also, could you please check if `ora` automatically adds a new line after failing (i.e, after calling `spinner.fail()`? At least if we don't fail, no new line will be added and the console will log like this:

![log](https://i.ibb.co/P9bJNYd/2019-04-26-T18-28-39-958x314.png)